### ISSUE TITLE
ITE-10: support multiple predicates per step?

### ISSUE THREAD

**Issue by adityasaky** [in-toto/ITE#58]
The current draft for ITE-10 supports multiple predicate types per step, but it's possible this introduces more complexity than is needed. The original idea was to support witness's attestation collections-like workflows, but we may be better off specially handling those types or finding some other mechanism for such steps.

**Comment by alanssitis** [comment]
IMO, this scenario seems more suitable for something like sub-policies. Although we could do something specific for the collections and handle them differently.

**Comment by adityasaky** [comment]
Yeah, I'm leaning towards specialcasing collections. There's a PoC of this on attestation-verifier: https://github.com/in-toto/attestation-verifier/pull/26
